### PR TITLE
feat(newProject): Use withExperiment HoC

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Also cleaned up some of the internal API of this component a bit. There was a lot of parameter passing going on.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/projectInstall/createProject.jsx [hunk 1]
```diff
@@ -66,47 +67,33 @@ class CreateProject extends React.Component {
...
 
   componentDidMount() {
-    const {organization} = this.props;
-    const alertDefaultsExperimentVariant =
-      organization.experiments?.AlertDefaultsExperiment;
-    const isInAlertDefaultsExperiment = [
-      '2OptionsV1',
-      '3OptionsV2',
-      'controlV1',
-    ].includes(alertDefaultsExperimentVariant);
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/projectInstall/createProject.jsx:76]: These are the only variants, no need for this check, a user is always in the experiment.

File: src/sentry/static/sentry/app/views/projectInstall/createProject.jsx [hunk 2]
```diff
@@ -66,47 +67,33 @@ class CreateProject extends React.Component {
...
-
-    if (isInAlertDefaultsExperiment) {
-      logExperiment({
-        organization,
-        key: 'AlertDefaultsExperiment',
-      });
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/projectInstall/createProject.jsx:82]: Done by the HoC.

File: src/sentry/static/sentry/app/views/projectInstall/createProject.jsx [hunk 3]
```diff
@@ -332,11 +319,11 @@ class CreateProject extends React.Component {
                for your API server and frontend client.`
             )}
           </HelpText>
-          {hasIssueAlertOptionsEnabled && (
+          {this.issueAlertOptionsEnabled && (
             <PageHeading withMargins>{t('Choose a platform')}</PageHeading>
           )}
           <PlatformPicker platform={platform} setPlatform={this.setPlatform} showOther />
-          {hasIssueAlertOptionsEnabled && (
+          {this.issueAlertOptionsEnabled && (
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/projectInstall/createProject.jsx:326]: getters aren't cached right? Perf impact on this getter is negligible (I can't even tell if all of these are in the same method either), but something to be aware of.

**Review by billyvg — APPROVED** [review]

**Comment by billyvg** [comment]
Hmm there are some snapshot failures that indicate behavior has been changed

File: src/sentry/static/sentry/app/views/projectInstall/createProject.jsx (hunk 3 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/projectInstall/createProject.jsx:326]: Correct it is not cached.

**Comment by EvanPurkhiser** [comment]
Yeah taking a look at snapshots now.

File: src/sentry/static/sentry/app/views/projectInstall/createProject.jsx [hunk 4]
```diff
@@ -66,47 +67,40 @@ class CreateProject extends React.Component {
...
   }
 
-  renderProjectForm = (
-    projectName,
-    team,
-    teams,
-    platform,
-    hasIssueAlertOptionsEnabled,
-    organization,
-    canSubmitForm
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/projectInstall/createProject.jsx:103]: I prefer pulling things from state / props where they're needed rather than passing parameters around.  This made it pretty hard to follow what was going on here.... so I just removed all of them.

File: src/sentry/static/sentry/app/views/projectInstall/createProject.jsx [hunk 5]
```diff
@@ -66,47 +67,40 @@ class CreateProject extends React.Component {
...
+      organization_id: organization.id,
+      alert_defaults_experiment_variant: experimentAssignment,
+    });
+  }
+
+  get issueAlertOptionsEnabled() {
+    return ['2OptionsV1', '3OptionsV2'].includes(this.props.experimentAssignment);
   }
 
-  renderProjectForm = (
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/projectInstall/createProject.jsx:96]: Didn't need to be a bound function property.

File: src/sentry/static/sentry/app/views/projectInstall/createProject.jsx [hunk 6]
```diff
@@ -66,47 +67,31 @@ class CreateProject extends React.Component {
...
+    trackAnalyticsEvent({
       eventKey: 'new_project.visited',
       eventName: 'New Project Page Visited',
-      organization_id: parseInt(this.props.organization.id, 10),
-    };
-    if (isInAlertDefaultsExperiment) {
-      analyticsEventOptions.alert_defaults_experiment_variant = alertDefaultsExperimentVariant;
-    }
-    trackAnalyticsEvent(analyticsEventOptions);
+      organization_id: organization.id,
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/projectInstall/createProject.jsx:75]: organization_id is coerced in getsentry from string to int.

File: src/sentry/static/sentry/app/views/projectInstall/createProject.jsx [hunk 7]
```diff
@@ -241,11 +231,9 @@ class CreateProject extends React.Component {
         ruleId = ruleData.id;
       }
       this.trackIssueAlertOptionSelectedEvent(
-        organization,
         projectData,
         defaultRules,
         shouldCreateCustomRule,
-        platform,
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/projectInstall/createProject.jsx:248]: This was wrong. This parameter was not part of the `trackIssueAlertOptionSelectedEvent` signature.

**Review by billyvg — APPROVED** [review]
